packages/blade/src/components/Toast/ToastContainer.web.tsx

```diff
@@ -0,0 +1,271 @@
+  const CONTAINER_OFFSET = isMobile ? 16 : 24;
+
+  const infoToasts = React.useMemo(() => toasts.filter((toast) => !isPromotionalToast(toast)), [
+    toasts,
+  ]);
+  const promoToasts = React.useMemo(() => toasts.filter((toast) => isPromotionalToast(toast)), [
+    toasts,
+  ]);
+
+  // always keep promo toasts at the bottom of the stack
+  const recomputedToasts = React.useMemo(() => [...infoToasts, ...promoToasts], [
+    infoToasts,
+    promoToasts,
+  ]);
+
+  const hasPromoToast = promoToasts.length > 0 && promoToasts[0]?.visible;
+  const promoToastHeight = promoToasts[0]?.height ?? 0;
+  const isExpanded = hasManuallyExpanded || recomputedToasts.length <= MIN_TOASTS;
+
+  React.useEffect(() => {
+    if (hasManuallyExpanded) {
+      handlers.startPause();
+    } else {
+      handlers.endPause();
+    }
+  }, [handlers, hasManuallyExpanded]);
+
+  React.useLayoutEffect(() => {
+    // find the first toast which is visible
+    setFrontToastHeight(infoToasts.find((t, index) => t.visible && index === 0)?.height ?? 0);
+  }, [infoToasts]);
+
+  // calculate total height of all toasts
+  const totalHeight = React.useMemo(() => {
+    return (
+      recomputedToasts
+        // only consider visible recomputedToasts
+        .filter((toast) => toast.visible)
+        .reduce((prevHeight, toast) => prevHeight + (toast.height ?? 0), 0) +
+      recomputedToasts.length * CONTAINER_OFFSET
+    );
+  }, [recomputedToasts, CONTAINER_OFFSET]);
+
+  const calculateYPosition = React.useCallback(
+    ({ toast, reverseOrder = false, index, defaultPosition }: CalculateYPositionProps) => {
+      const relevantToasts = infoToasts.filter(
+        (t) => (t.position ?? defaultPosition) === (toast.position ?? defaultPosition) && t.height,
+      );
+      const toastIndex = relevantToasts.findIndex((t) => t.id === toast.id);
+      // number of toasts before this toast
+      const toastsBefore = relevantToasts.filter((toast, i) => i < toastIndex && toast.visible)
+        .length;
+
+      let scale = index < MAX_TOASTS ? 1 : Math.max(0.7, 2 - (toastsBefore * SCALE_FACTOR + 1));
+      // y position of toast,
+      let offset = relevantToasts
+        .filter((toast) => toast.visible)
+        .slice(...(reverseOrder ? [toastsBefore + 1] : [0, toastsBefore]))
+        .reduce((acc, toast) => {
+          if (isExpanded) {
+            return acc + (toast.height ?? 0) + GUTTER;
+          }
+          return acc + PEEK_GUTTER;
+        }, 0);
+
+      // lift all info toasts up if there is a promo toast
+      if (hasPromoToast) {
+        offset += GUTTER + promoToastHeight;
+      }
+      // promo toasts should always be on bottom
+      if (isPromotionalToast(toast)) {
+        offset = 0;
+        scale = 1;
+      }
+
+      return { offset, scale: isExpanded ? 1 : scale };
+    },
+    [hasPromoToast, infoToasts, isExpanded, promoToastHeight],
+  );
+
+  return (
+    <BaseBox
+      position="fixed"
+      zIndex={9999}
```

Comment:
Why this zIndex? Store it as a constant somewhere so that we can standardise this now

Reply:
Done